pkg/controller/compute/kubernetes/azure_handler.go

```diff
@@ -33,15 +34,17 @@ type AKSClusterHandler struct{}
 
 // Find AKSCluster resource
 func (r *AKSClusterHandler) Find(name types.NamespacedName, c client.Client) (corev1alpha1.Resource, error) {
-	instance := &azurecomputev1alpha1.AKSCluster{}
-	err := c.Get(ctx, name, instance)
-	return instance, err
+	instance := &v1alpha1.AKSCluster{}
+	if err := c.Get(ctx, name, instance); err != nil {
+		return nil, errors.Wrapf(err, "failed to retrieve %s: %s", v1alpha1.AKSClusterKind, name)
+	}
+	return instance, nil
```

Comment:
`errors.Wrap()` and friends will return `nil` when passed a `nil` error, so it's not strictly necessary to test `err != nil`. You could instead write:

```go
return instance, errors.Wrapf(c.Get(ctx, name, instance), "failed to retrieve %s: %s", v1alpha1.AKSClusterKind, name)
```

That said, I'm not sure I prefer this "one liner" style. The line gets a bit long, and it always returns a pointer (rather than nil) even if there is an error. Should not matter in practice, but is a little unusual.

Reply:
Ah, yes. There is an "edge case", where I prefer to return `nil` if there is an error.

Reply:
The latter part is a deal-breaker for me, I prefer to return `nil` (in this specific case, it doesn't really matter), on other cases, however, this alleviates a need to reset the result to `nil` in the client call.